File: src/utils/__tests__/units.test.ts
import { kgToLb, lbToKg, roundTo } from '@/utils/units';

describe('unit conversion', () => {
  it('converts kg to lb and back', () => {
    expect(kgToLb(100)).toBeCloseTo(220.462, 3);
    expect(lbToKg(220.462)).toBeCloseTo(100, 3);
  });
  it('round-trips without drift', () => {
    expect(lbToKg(kgToLb(60))).toBeCloseTo(60, 6);
  });
  it('rounds to a given precision', () => {
    expect(roundTo(220.4622, 1)).toBe(220.5);
    expect(roundTo(1.2345, 2)).toBe(1.23);
  });
});

File: android-native/app/src/test/java/com/fitnessrpg/app/util/UnitsTest.kt
package com.fitnessrpg.app.util

import org.junit.Assert.assertEquals
import org.junit.Test

class UnitsTest {
    @Test
    fun `converts kg to lb and back`() {
        assertEquals(220.462, kgToLb(100.0), 1e-3)
        assertEquals(100.0, lbToKg(220.462), 1e-3)
    }

    @Test
    fun `round-trips without drift`() {
        assertEquals(60.0, lbToKg(kgToLb(60.0)), 1e-6)
    }

    @Test
    fun `rounds to a given precision`() {
        assertEquals(220.5, roundTo(220.4622, 1), 1e-9)
        assertEquals(1.23, roundTo(1.2345, 2), 1e-9)
    }
}
